gitpod-io/gitpod
# [image-builder-mk3] Improve error handling

## Description
This PR 
- improves the error handling of image-builder-mk3. When resolving base-images against an authenticated registry, it now returns the correct error. Same goes for any other error that might happen during image resolution.
  For example: trying to pull from a registry the user is not authenticated against: <img width="546" alt="image" src="https://user-images.githubusercontent.com/3210701/166820197-662b9670-19a2-4a8d-a7a3-f666b9211aab.png">

- introduces a basic set of unit tests for this behaviour.
- adds support for pre-caching references from registries which require authentication.

## How to test
- see unit tests
- try and start a workspace which attempts to pull from registry which requires authentication, without providing this authentication.

## Release Notes

```release-note
[image-builder] Improve error handling (no more "hostname required")
```
